src/bin/gpfdist/gpfdist.c

```diff
@@ -836,6 +848,37 @@ static void parse_command_line(int argc, const char* const argv[],
 			exit(1);
         }
     }
+
+	/* validate opt.I and opt.O */
+	if (opt.I || opt.O)
```

Comment:
can we delete this `if` statment to don't check opt.I and opt.O twince?
```c
    extern const char* validate_transform_opt(struct transform* trlist, const char* name);
	if (opt.I)
	{
		char *p = gstring_trim(apr_pstrdup(pool, opt.I));
		const char *err = validate_transform_opt(opt.trlist, p);

		if (err)
			usage_error(err, 0);

		opt.I = p;
	}

	if (opt.O)
	{
		char *p = gstring_trim(apr_pstrdup(pool, opt.O));
		const char *err = validate_transform_opt(opt.trlist, p);

		if (err)
			usage_error(err, 0);

		opt.O = p;
	}
```

Reply:
In this example extern function is always defined, even if it's not used later.
Another reason was to avoid, not all, but a bit of copypaste.

Anyway, I think it's like "exchange an awl for soap" here.

Reply:
I think than there is no problem when a function is defined more global. If we have less checks, the code become simpler. And some microseconds faster)

Reply:
This is more not about a checks, but about code readability. Placing `extern` inside if condition informs the developer, that this function is used only in this block. The code above has example of the same situation. If if we put transform_config() out of if condition, then developer can't say that function is used only in next condition and should scroll below (to our new flags, or even lower to future changes) to check it.

From another points, it's awl for soap.